src/ProjectService.Validation/CreateProjectValidator.cs

```diff
@@ -1,91 +1,179 @@
-            });
-
-            When(
-                news => news.DepartmentId.HasValue,
-                () =>
-                    RuleFor(news => news.DepartmentId)
-                        .Must(DepartmentId => DepartmentId != Guid.Empty)
-                        .WithMessage("Wrong type of department Id."));
+              try
+              {
+                var byteString = new Span<byte>(new byte[images.Content.Length]);
+                return Convert.TryFromBase64String(images.Content, byteString, out _);
+              }
+              catch
+              {
+                return false;
+              }
+            }).WithMessage("Wrong image content.");
+      });
+
+      When(project => !string.IsNullOrEmpty(project.ShortDescription?.Trim()), () =>
+      {
+        RuleFor(project => project.ShortDescription)
+          .MaximumLength(300)
+          .WithMessage("Project short description is too long.");
+      });
+
+      When(project => !string.IsNullOrEmpty(project.Description?.Trim()), () =>
+      {
+        RuleFor(project => project.Description)
+          .MaximumLength(300)
+          .WithMessage("Project description is too long.");
+      });
+
+      When(
+        news => news.DepartmentId.HasValue,
+        () =>
+          RuleFor(news => news.DepartmentId)
+            .Must(DepartmentId => DepartmentId != Guid.Empty)
+            .WithMessage("Wrong type of department Id."));
+    }
+
+    private bool CheckDepartmentExistence(Guid? departmentId)
+    {
+      if (!departmentId.HasValue)
+      {
+        return true;
+      }
+
+      string logMessage = "Department with id: {id} not found.";
+
+      try
+      {
+        var response = _rcCheckDepartmentsExistence.GetResponse<IOperationResult<ICheckDepartmentsExistence>>(
+          ICheckDepartmentsExistence.CreateObj(new List<Guid> { departmentId.Value })).Result;
+        if (response.Message.IsSuccess)
+        {
+          if (!response.Message.Body.DepartmentIds.Any())
+          {
+            return false;
+          }
+          return true;
+        }
+
+        _logger.LogWarning("Can not find department with this Id: {departmentId}: " +
+          $"{Environment.NewLine}{string.Join('\n', response.Message.Errors)}");
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, logMessage);
+      }
+
+      return false;
+    }
+
+    private bool CheckUserExistence(List<Guid> userIds)
+    {
+      if (!userIds.Any())
+      {
+        return true;
+      }
+
+      string logMessage = "Cannot check existing users withs this ids {userIds}";
+
+      try
+      {
+        var response = _rcCheckUsersExistence.GetResponse<IOperationResult<ICheckUsersExistence>>(
+          ICheckUsersExistence.CreateObj(userIds)).Result;
+        if (response.Message.IsSuccess)
+        {
+          return true;
         }
+
+        _logger.LogWarning("Can not find {userIds} with this Ids: {userIds}: " +
```

Comment:
here is example of LogWarning:
_logger.LogWarning("Can not add certificate image to certificate. Reason: '{Errors}'",
                        string.Join(',', response.Message.Errors));
Check arguments.

Reply:
done